create.cjs

```diff
@@ -701,3 +676,85 @@ function generateDonwloadStats() {
     ).join('\n')
   );
 }
+
+/**
+ * @param {VirtualPackage} pkg
+ */
+async function writePackage(pkg) {
+  await fsPromises.mkdir(pkg.path, { recursive: true });
+  let hasChanged = false;
+
+  const promises = [];
+
+  const ps = new PathScurry(pkg.path);
+
+  const files = await ps.walk({
+    filter(file) {
+      return !['dist', 'node_modules', 'package.json'].includes(file.name) && file.name[0] !== '.';
+    }
+  });
+
+  const existingFiles = files
+    .filter(f => f.isFile())
+    .map(file => file.relativePosix());
+
+  const extraFiles = existingFiles.filter(file => !(file in pkg.files));
+
+  const packageJsonPath = path.join(pkg.path, 'package.json');
+  const existingPackageJson = fs.existsSync(packageJsonPath)
+    ? JSON.parse(await fsPromises.readFile(packageJsonPath, 'utf8'))
+    : {};
+
+  // remove extra files
+  if (extraFiles.length) {
+    extraFiles.forEach(file => {
+      promises.push(
+        fsPromises.rm(path.join(pkg.path, file))
+      );
+    });
+    hasChanged = true;
+  }
+
+  // write files, and check if they changed
+  Object.entries(pkg.files).forEach(([file, content]) => {
+    const filePath = path.join(pkg.path, file);
+    promises.push((async () => {
+      if (fs.existsSync(filePath)) {
```

Comment:
We could avoid the extra filesystem API call by re-using the `existingFiles` above.